src/extensionsIntegrated/CustomSnippets/helper.js

```diff
@@ -0,0 +1,778 @@
+        // abbreviation and template text are required fields
+        // they both should have some value only then save button will be enabled
+        const $abbrInput = $("#abbr-box");
+        const $templateInput = $("#template-text-box");
+
+        const $saveBtn = $("#save-custom-snippet-btn button");
+
+        // make sure that the required fields has some value
+        const hasAbbr = $abbrInput.val().trim().length > 0;
+        const hasTemplate = $templateInput.val().trim().length > 0;
+        $saveBtn.prop("disabled", !(hasAbbr && hasTemplate));
+    }
+
+    /**
+     * this function is responsible to get the current language context,
+     * from the editor at cursor position
+     *
+     * @param {Editor} editor - The editor instance
+     * @returns {string|null} - The language ID or null if not available
+     */
+    function getCurrentLanguageContext(editor) {
+        // first try to get the language at cursor pos
+        // if it for some reason fails, then just go for the file extension
+        try {
+            const language = editor.getLanguageForPosition();
+            const languageId = language ? language.getId() : null;
+            return languageId;
+        } catch (e) {
+            return getCurrentFileExtension(editor);
+        }
+    }
+
+    /**
+     * Gets the current file extension from the editor
+     * @param {Editor} editor - The editor instance
+     * @returns {string|null} - The file extension or null if not available
+     */
+    function getCurrentFileExtension(editor) {
+        const filePath = editor && editor.document && editor.document.file ? editor.document.file.fullPath : undefined;
+        if (filePath) {
+            return filePath.substring(filePath.lastIndexOf(".")).toLowerCase();
+        }
+        return null;
+    }
+
+    /**
+     * Checks if a snippet is supported in the given language context
+     * Falls back to file extension matching if language mapping isn't available
+     *
+     * @param {Object} snippet - The snippet object
+     * @param {string|null} languageContext - The current language context
+     * @param {Editor} editor - The editor instance for fallback
+     * @returns {boolean} - True if the snippet is supported
+     */
+    function isSnippetSupportedInLanguageContext(snippet, languageContext, editor) {
+        if (snippet.fileExtension.toLowerCase() === "all") {
+            return true;
+        }
+
+        if (languageContext) {
+            const effectiveExtension = mapLanguageToExtension(languageContext);
+
+            // if we have a proper mapping (starts with .), use language context matching
+            if (effectiveExtension.startsWith(".")) {
+                const supportedExtensions = snippet.fileExtension
+                    .toLowerCase()
+                    .split(",")
+                    .map((ext) => ext.trim());
+
+                return supportedExtensions.some((ext) => ext === effectiveExtension);
```

Comment:
these could be replaced with language check directly from `languageId = LanguageManager.getLanguageForPath(editor.document.file._path).getId();`